### DISCUSSION TITLE
Hook for job termination

### DISCUSSION THREAD

**Discussion by marcbelmont** [dagster-io/dagster#7464] [Q&A] [answered]
I want to call a function when a job is terminated to do some clean up.

1) start a job
2) Terminate the job with Terminate button on Dagit
3) The following event is in the logs: "Multiprocess executor: interrupted all active child processes"
4) How do I execute code after this event is triggered?

I've tried to use the failure_hook but it's not called for termination.

**Comment by yuhan** [comment, marked as answer]
Hi! You can use [`@run_status_sensor`](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors) to run something when a run is terminated (i.e. canceled). It's a special sensor that acts on [`DagsterRunStatus`](https://docs.dagster.io/_apidocs/internals#dagster.DagsterRunStatus) changes.

```python
from dagster import run_status_sensor, RunStatusSensorContext, DagsterRunStatus

@run_status_sensor(pipeline_run_status=DagsterRunStatus.CANCELED)
def do_something_on_run_canceled(context: RunStatusSensorContext):
   ...
```

**Comment by nvinhphuc** [comment]
I still think we need a hook for job/op termination. Using run_status_sensor requires us to define a sensor and a job which is not very convenient.

**Reply by nvinhphuc** [reply]
And termination seems like a good type of hooks, it's like a special failure_hook.
